pkg/templates/externalccm/packet.go

```diff
@@ -0,0 +1,277 @@
+				Resources: []string{"persistentvolumes"},
+				Verbs:     []string{"get", "list", "update", "watch"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"endpoints"},
+				Verbs:     []string{"create", "get", "list", "watch", "update"},
+			},
+		},
+	}
+}
+
+func packetClusterRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "ClusterRoleBinding",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "system:cloud-controller-manager",
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Name:     "system:cloud-controller-manager",
+			Kind:     "ClusterRole",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      digitaloceanSAName,
+				Namespace: metav1.NamespaceSystem,
+			},
+		},
+	}
+}
+
+func packetDeployment() *appsv1.Deployment {
+	var (
+		replicas int32 = 1
+	)
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      packetDeploymentName,
+			Namespace: metav1.NamespaceSystem,
+			Labels: map[string]string{
+				"app": "packet-cloud-controller-manager",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "packet-cloud-controller-manager",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"scheduler.alpha.kubernetes.io/critical-pod": "",
+					},
+					Labels: map[string]string{
+						"app": "packet-cloud-controller-manager",
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: packetSAName,
+					Tolerations: []corev1.Toleration{
+						{
+							Key:      "node-role.kubernetes.io/master",
+							Operator: corev1.TolerationOpExists,
+							Effect:   corev1.TaintEffectNoSchedule,
+						},
+						{
+							Key:    "node.cloudprovider.kubernetes.io/uninitialized",
+							Value:  "true",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+						{
+							Key:      "CriticalAddonsOnly",
+							Operator: corev1.TolerationOpExists,
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "packet-cloud-controller-manager",
+							Image: "packethost/packet-ccm:" + packetCCMVersion,
+							Command: []string{
+								"./packet-cloud-controller-manager",
+								"--cloud-provider=packet",
+								"--leader-elect=false",
+								"--allow-untagged-cloud=true",
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name: "PACKET_AUTH_TOKEN",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: credentials.SecretName,
+											},
+											Key: config.PacketAPIKey,
```

Comment:
This is referencing the old API. You should pick it from the `credentials` package.